### PR TITLE
feat: add baidu translation impl

### PR DESCRIPTION
**Pull request by SCMRCORE**
### Describe what this PR does / why we need it
The implementation of the Baidu translation component

### Does this pull request fix one issue?

#98 

### Describe how you did it
With Baidu translation Open APIs

### Describe how to verify it
I implemented a text translation and here are two examples:
success
文本翻译成功
failure
文本翻译异常情况

### CODE REVIEW

File: community/function-calling/spring-ai-alibaba-starter-function-calling-baidutranslate/src/main/resources/META-INF/spring/org.springframework.boot.autoconfigure.AutoConfiguration.imports
Comment (PolarishT) [inline community/function-calling/spring-ai-alibaba-starter-function-calling-baidutranslate/src/main/resources/META-INF/spring/org.springframework.boot.autoconfigure.AutoConfiguration.imports:1]: please add space

**Review by PolarishT — CHANGES_REQUESTED** [review]
please use maven profile: license env to build license header

**Comment by SCMRCORE** [comment]
> 请使用 maven 配置文件：许可证环境构建许可证标头

Thanks, I changed the other problems first, and when you merge your pr, I can use the plugin to change the license. QAQ

File: community/function-calling/spring-ai-alibaba-starter-function-calling-baidutranslate/src/main/java/com/alibaba/cloud/ai/functioncalling/baidutranslate/BaidutranslateService.java [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+		body.add("sign", sign);
+		return body;
+	}
+
+	private Response parseResponse(String responseData) {
+
+		Gson gson = new Gson();
+		JsonElement jsonElement = gson.fromJson(responseData, JsonElement.class);
+
+		if (jsonElement.getAsJsonObject().has("error_code")) {
```
Comment (yuluo-yx) [inline community/function-calling/spring-ai-alibaba-starter-function-calling-baidutranslate/src/main/java/com/alibaba/cloud/ai/functioncalling/baidutranslate/BaidutranslateService.java:108]: 这个判断是不是不太严谨？没有 code 之类的吗？

File: community/function-calling/spring-ai-alibaba-starter-function-calling-baidutranslate/src/main/java/com/alibaba/cloud/ai/functioncalling/baidutranslate/TranslationResponse.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.alibaba.cloud.ai.functioncalling.baidutranslate;
+
+import java.util.List;
+
+class TranslationResponse {
```
Comment (yuluo-yx) [inline community/function-calling/spring-ai-alibaba-starter-function-calling-baidutranslate/src/main/java/com/alibaba/cloud/ai/functioncalling/baidutranslate/TranslationResponse.java:21]: Request 都用了 record 这里是不是可以和 request 一起使用 record type 定义？

File: community/function-calling/spring-ai-alibaba-starter-function-calling-baidutranslate/src/main/java/com/alibaba/cloud/ai/functioncalling/baidutranslate/TranslationResult.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.alibaba.cloud.ai.functioncalling.baidutranslate;
+
+class TranslationResult {
```
Comment (yuluo-yx) [inline community/function-calling/spring-ai-alibaba-starter-function-calling-baidutranslate/src/main/java/com/alibaba/cloud/ai/functioncalling/baidutranslate/TranslationResult.java:19]: ditto

File: community/function-calling/spring-ai-alibaba-starter-function-calling-baidutranslate/src/main/java/com/alibaba/cloud/ai/functioncalling/baidutranslate/BaidutranslateService.java [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+		body.add("to", request.to);
+		body.add("appid", appId);
+		body.add("salt", salt);
+		body.add("sign", sign);
+		return body;
+	}
+
+	private Response parseResponse(String responseData) {
+
+		Gson gson = new Gson();
```
[the remark this answers is not in the document]
Comment (SCMRCORE) [reply community/function-calling/spring-ai-alibaba-starter-function-calling-baidutranslate/src/main/java/com/alibaba/cloud/ai/functioncalling/baidutranslate/BaidutranslateService.java:105]: Sorry, before developing this pluging, I studied the TranslateService.java and used this

File: community/function-calling/spring-ai-alibaba-starter-function-calling-baidutranslate/src/main/java/com/alibaba/cloud/ai/functioncalling/baidutranslate/BaidutranslateService.java (hunk 1 above)
Comment (SCMRCORE) [reply community/function-calling/spring-ai-alibaba-starter-function-calling-baidutranslate/src/main/java/com/alibaba/cloud/ai/functioncalling/baidutranslate/BaidutranslateService.java:108]: I originally wanted to use one class/record to map this response, but Baidu Translate API gave the response  belike:
百度翻译响应
so, I use " if " and return the error_code and error_msg. 
And If there's a better way , I will correct this part  ,please tell me thanks. QAQ

Comment (yuluo-yx) [reply community/function-calling/spring-ai-alibaba-starter-function-calling-baidutranslate/src/main/java/com/alibaba/cloud/ai/functioncalling/baidutranslate/BaidutranslateService.java:108]: imo, 可以先拿到 errcode 的值，然后和错误码比较一下，不是判断是不是有这个字段存在

可以先判断此次 http 请求是否成功（response.code == 200?）
然后在判断业务系统的响应（errCode == custom_err_code?）

Comment (SCMRCORE) [reply community/function-calling/spring-ai-alibaba-starter-function-calling-baidutranslate/src/main/java/com/alibaba/cloud/ai/functioncalling/baidutranslate/BaidutranslateService.java:108]: Thank you for your guidance !

File: community/function-calling/spring-ai-alibaba-starter-function-calling-baidutranslate/src/main/java/com/alibaba/cloud/ai/functioncalling/baidutranslate/BaidutranslateService.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (PolarishT) [reply community/function-calling/spring-ai-alibaba-starter-function-calling-baidutranslate/src/main/java/com/alibaba/cloud/ai/functioncalling/baidutranslate/BaidutranslateService.java:105]: 那就全部使用Gson？

[the remark this answers is not in the document]
Comment (SCMRCORE) [reply community/function-calling/spring-ai-alibaba-starter-function-calling-baidutranslate/src/main/java/com/alibaba/cloud/ai/functioncalling/baidutranslate/BaidutranslateService.java]: Have changed the class to record and changed gson to jackson. Thanks for guiding

**Review by PolarishT — APPROVED** [review]
LGTM

**Pull request merged**